src/cmp/comparison.rs

```diff
@@ -15,15 +16,24 @@ pub enum Comparison {
 
 #[derive(Debug, PartialEq, Eq)]
 pub enum Diff {
```

Comment:
An alternative I had considered is having a separate `Diff` enum. Thus, all these `#[cfg(...` macros would be avioded, but then there would be code duplication in everything related `Contents`. I thought that current implenetation is better.